common/common.cpp

```diff
@@ -804,10 +804,10 @@ std::vector<llama_token> llama_tokenize(
     // upper limit for the number of tokens
     int n_tokens = text.length() + add_bos;
     std::vector<llama_token> result(n_tokens);
-    n_tokens = llama_tokenize(ctx, text.c_str(), result.data(), result.size(), add_bos);
+    n_tokens = llama_tokenize(ctx, text.c_str(), text.length(), result.data(), result.size(), add_bos);
```

Comment:
Why is this needed?

Reply:
Otherwise I get these 

```
main : error: token 3 detokenizes to ><(1) but tokenization of this detokenizes to ><(0)
main : error: codepoint 0 detokenizes to ><(0) instead of ><(1)
```

errors from `test-tokenizer-1`. C strings do not differentiate "\0" from "\0\0".

Reply:
But should that be possible to run into in practice? For UTF-8 text, the NUL byte should only appear at the end of a string. Who would be trying to pass binary data to the tokenizer?

Reply:
> But should that be possible to run into in practice? 

I'm not sure, but these test cases work for the `sentencepiece` tokenizer.

Reply:
> For UTF-8 text, the NUL byte should only appear at the end of a string.

Not sure if this is correct.